http-builder-ng-core/src/main/java/groovyx/net/http/HttpObjectConfig.java

```diff
@@ -197,6 +198,27 @@
          * @return true if cookies are enabled, false if not
          */
         boolean getCookiesEnabled();
+
+        /**
+         * A `Consumer<Object>` may be provided, which will have the internal client implementation reference passed into it to allow further
+         * client configuration beyond what it supported directly by HttpBuilder-NG. The `Object` passed in will be an instance of the internal client
+         * builder type, not necessarily the client itself.
+         *
+         * This configuration method should _only_ be used when the desire configuration is _not_ available directly through the `HttpBuilder`
+         * configuration interfaces. Configuring in this manner may override helpful configuration already applied by the library.
+         *
+         * Note that a Groovy closure may be used to replace the `Consumer` with no modification of functionality.
+         *
+         * This operation is optional. If a client-implementation does not support it, an {@link UnsupportedOperationException} will be thrown.
+         */
+        void customizeClient(Consumer<Object> customizer);
```

Comment:
Also, can we add a clientCustomizer(Closure) method as well (and just wrap/delegate to the original method)? It tends to make usage more natural when using TypeChecked or CompileStatic. A default method on the interface would do the job.